base/range.jl

```diff
@@ -1087,3 +1087,13 @@ julia> mod(3, 0:2)
 """
 mod(i::Integer, r::OneTo) = mod1(i, last(r))
 mod(i::Integer, r::AbstractUnitRange{<:Integer}) = mod(i-first(r), length(r)) + first(r)
+
+"""
+    clamp(x::Integer, r::AbstractUnitRange)
+
+Clamp `x` to lie within range `r`.
+
+!!! compat "Julia 1.6"
+     This method requires at least Julia 1.6.
+"""
+clamp(x::Integer, r::AbstractUnitRange{<:Integer}) = clamp(x, first(r), last(r))
```

Comment:
It's a bit weird to me that the convenient version of `clamp` and `mod` doesn't support `x::Real`.

```julia
julia> clamp(1.5, -1, 3)
1.5
```

Reply:
I don't really think it's weird, since `1.5` is clearly not in `-1:3`. I think it would be much more awkward if `UnitRange`s behaved like floating point intervals only in some contexts.